src/physics/euler.cpp

```diff
@@ -359,6 +361,7 @@ ::compute_sound ( const std::array<real,nstate> &conservative_soln ) const
 {
     real density = conservative_soln[0];
     if(density<0.0) {
+        std::cout << "WARNING: Negative density encountered! Setting density to BIG_NUMBER." << std::endl;
```

Comment:
Do you think it would be worth it to move these checks to a separate function `check_positive_density(const real density)`? Since these are repeated in euler.cpp quite frequently.

Reply:
fixed

Reply:
Thanks for adding those!